src/commands/mod.rs

```diff
@@ -289,6 +289,58 @@ impl DaemonControl {
         GetAddressResult::new(address)
     }
 
+    /// list addresses
+    pub fn list_addresses(
+        &self,
+        start_index: Option<u32>,
+        count: Option<u32>,
+    ) -> ListAddressesResult {
+        let mut db_conn = self.db.connection();
+        let receive_index: u32 = db_conn.receive_index().into();
+        let change_index: u32 = db_conn.change_index().into();
+
+        let start_index = start_index.unwrap_or(0);
+
+        let index = if let Some(c) = count {
+            (start_index + c).saturating_sub(1)
+        } else {
+            let max_index = receive_index.max(change_index);
+            if start_index >= max_index {
+                start_index
+            } else {
+                max_index.saturating_sub(1)
```

Comment:
Same here. Use `checked_sub` and error if it returns None instead of rolling over.

Reply:
Actually why do you substract `1` at all here? This is already the max index, not a count.

I think this can simply be `std::max(start_index, max_index)`.